### PR TITLE
MULE-18546: Change lookup policy for plugin classloaders regarding org.mule.sdk package

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

**Comment by ndinu** [comment]
--validate

File: modules/container/src/main/java/org/mule/runtime/container/internal/ContainerClassLoaderFactory.java [hunk 1]
```diff
@@ -174,16 +179,31 @@ protected ArtifactClassLoader createArtifactClassLoader(final ClassLoader parent
...
   }
 
+  /**
+   * Returns an {@link Optional} with the {@link LookupStrategy} if the one to use for the exportedPackage is other than
+   * a {@link ContainerOnlyLookupStrategy}
+   *
+   * @param exportedPackage name of the package
+   * @return
+   */
+  private Optional<LookupStrategy> getSpecialLookupStrategy(String exportedPackage) {
```
Comment (marianogonzalez) [inline modules/container/src/main/java/org/mule/runtime/container/internal/ContainerClassLoaderFactory.java:196]: Usage of Optional is overkill here. Just return the policy to be used

File: tests/test-extensions/classloading-extension/src/main/java/org/mule/sdk/api/runtime/connectivity/ReconnectionCallback.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.sdk.api.runtime.connectivity;
+
+import org.mule.runtime.api.connection.ConnectionException;
+
+public interface ReconnectionCallback {
```
[the remark this answers is not in the document]
Comment (ndinu) [reply tests/test-extensions/classloading-extension/src/main/java/org/mule/sdk/api/runtime/connectivity/ReconnectionCallback.java:11]: This is just a test class that colides with one that the runtime has whose package begins with org.mule.sdk.api.

[the remark this answers is not in the document]
Comment (ndinu) [reply tests/test-extensions/classloading-extension/src/main/java/org/mule/sdk/api/runtime/connectivity/ReconnectionCallback.java:11]: This is asociated with a test added in mule-integration-tests to check that this is the class used in the extension classloader.

**Pull request merged**